### **📌 Objective – Level 2, Task 1: Table Booking & Online Delivery**  

In this task, we aim to **analyze the impact of table booking and online delivery on restaurant performance**.  

### **🎯 Key Goals:**  
1️⃣ **Determine the percentage of restaurants** that offer **table booking** and **online delivery**.  
2️⃣ **Compare the average ratings** of restaurants with and without **table booking**.  
3️⃣ **Analyze how online delivery availability varies across different price ranges**.  

This analysis will help us understand:  
✔ How many restaurants provide these services.  
✔ Whether table booking affects restaurant ratings.  
✔ If online delivery is more common in high-end or budget restaurn you're ready! 😊

In [3]:
import pandas as pd
df = pd.read_csv('leve1_dataset.csv')

### 📊 Step 1: Calculate the Percentage of Restaurants Offering These Services

In [6]:
# Calculate the percentage of restaurants offering table booking and online delivery
table_booking_percentage = (df['Has Table booking'].sum() / len(df)) * 100
online_delivery_percentage = (df['Has Online delivery'].sum() / len(df)) * 100

# Print results
print(f"Percentage of restaurants offering table booking: {table_booking_percentage:.2f}%")
print(f"Percentage of restaurants offering online delivery: {online_delivery_percentage:.2f}%")

Percentage of restaurants offering table booking: 12.12%
Percentage of restaurants offering online delivery: 25.66%


#### 📊 Insights from Step 1:
✔ Only 12.12% of restaurants offer table booking – This suggests that most restaurants operate without reservations.

✔ 25.66% of restaurants provide online delivery – Online delivery is more common than table booking, showing that customers prefer convenience.



### 📌 Next Step: Step 2 – Compare Ratings of Restaurants with and without Table Booking
Now, let’s analyze whether restaurants with table booking receive higher ratings.

In [10]:
# Compare the average ratings of restaurants with and without table booking
ratings_comparison = df.groupby('Has Table booking')['Aggregate rating'].mean()

# Print results
print(ratings_comparison)


Has Table booking
0    2.559359
1    3.441969
Name: Aggregate rating, dtype: float64


#### 📊 Insights from Step 2: Table Booking vs. Ratings
✔ Restaurants without table booking (0) have an average rating of 2.56.

✔ Restaurants with table booking (1) have a higher average rating of 3.44.

🔹 Key Takeaways:

✅ Restaurants offering table booking receive better ratings.

✅ This suggests that customers prefer making reservations, and higher-end restaurants may provide better service.

✅ Restaurants without table booking tend to have lower ratings, possibly due to crowded spaces or inconsistent service.

### Next Step: Step 3 – Analyze Online Delivery Availability by Price Range
Now, let’s check whether online delivery is more common in low-cost or high-cost restaurants.

In [14]:
# Compare online delivery availability among different price ranges
delivery_by_price = df.groupby('Price range')['Has Online delivery'].mean() * 100

# Print results
print(delivery_by_price)


Price range
1    15.774077
2    41.310633
3    29.190341
4     9.044369
Name: Has Online delivery, dtype: float64


### **📊 Insights from Step 3: Online Delivery Availability by Price Range**  

✔ **15.77% of budget restaurants (Price Range 1) offer online delivery.**  
✔ **41.31% of mid-range restaurants (Price Range 2) offer online delivery – the highest percentage!**  
✔ **29.19% of premium restaurants (Price Range 3) offer online delivery.**  
✔ **Only 9.04% of high-end restaurants (Price Range 4) provide online delivery.**  

---

### **🔹 Key Takeaways:**  
✅ **Mid-range restaurants (Price Range 2) offer online delivery the most** – they likely target customers who prefer convenience.  
✅ **High-end restaurants (Price Range 4) rarely provide online delivery** – these places focus on in-house dining experiences.  
✅ **Budget restaurants (Price Range 1) have limited online delivery services**, possibly due to operational conyou want to proceed! 😊

### **📊 Level 2 – Task 1: Table Booking & Online Delivery**  

---

### **📌 Objective**  
- Determine the **percentage of restaurants** offering **table booking** and **online delivery**.  
- Compare the **average ratings** of restaurants with and without **table booking**.  
- Analyze how **online delivery availability** varies across **different price ranges**.  

---

### **1️⃣ Step 1: Percentage of Restaurants Offering Table Booking & Online Delivery**  
- **Table Booking:** 12.12% of restaurants offer table booking.  
- **Online Delivery:** 25.66% of restaurants provide online delivery.  
- **Insight:** Online delivery is more common than table booking, indicating a shift toward **convenience and takeout services**.  

---

### **2️⃣ Step 2: Comparing Ratings of Table Booking vs. Non-Table Booking Restaurants**  
- **Restaurants without table booking have an average rating of** **2.56**.  
- **Restaurants with table booking have an average rating of** **3.44**.  
- **Insight:** Restaurants offering table booking tend to have **higher ratings**, suggesting that **reservation-based dining improves customer experience**.  

---

### **3️⃣ Step 3: Online Delivery Availability by Price Range**  
- **Price Range 1 (Budget Restaurants):** 15.77% offer online delivery.  
- **Price Range 2 (Mid-Range Restaurants):** 41.31% offer online delivery (**highest percentage**).  
- **Price Range 3 (Premium Restaurants):** 29.19% offer online delivery.  
- **Price Range 4 (High-End Restaurants):** Only 9.04% offer online delivery.  
- **Insight:** Online delivery is **most common in mid-range restaurants**, while **high-end restaurants focus on dine-in experiences**.  

---

### **🎯 Final Summary**  
✔ **Successfully completed Level 2 – Task 1: Table Booking & Online Delivery.**  
✔ **Extracted key insights on table bookinmknow if you’re ready to start Task 2!** 😊